python/lsst/ip/isr/brighterFatterKernel.py

```diff
@@ -333,6 +333,17 @@ def toDict(self):
         outDict['rawMeans'] = {amp: np.array(self.rawMeans[amp]).tolist() for amp in self.rawMeans}
         outDict['rawVariances'] = {amp: np.array(self.rawVariances[amp]).tolist() for amp in
                                    self.rawVariances}
+
+        for amp in self.rawXcorrs.keys():
+            # Check to see if we need to repack the data.
+            correlationShape = np.array(self.rawXcorrs[amp]).shape
+            if nObs != correlationShape[0]:
+                if correlationShape[0] == np.sum(self.expIdMask[amp]):
+                    # Repack data.
+                    self.repackCorrelations(amp, correlationShape)
+                else:
+                    raise ValueError("Could not coerce rawXcorrs into appropriate shape.")
```

Comment:
Should the error be more informative?

Reply:
I've updated the message to note how many covariances we have (`correlationShape[0]`), and how many we expect from the mask (`np.sum(self.expIdMask[amp])`).  I think this will be a rare/never called error, as I think we'll either have all the entries (and need to mask them) or we'll have the pre-masked set (and will need to repad here).